evap/rewards/tools.py

```diff
@@ -71,14 +84,21 @@ def grant_reward_points(user, semester):
     required_courses = Course.objects.filter(participants=user, semester=semester, is_required_for_reward=True)
     if not required_courses.exists():
         return False
-    # does the user not participate in any more required courses in this semester?
-    if required_courses.exclude(voters=user).exists():
-        return False
-    # did the user not already get reward points for this semester?
-    if RewardPointGranting.objects.filter(user_profile=user, semester=semester).exists():
+
+    voted_courses_count = required_courses.filter(voters=user).count()
```

Comment:
[optional] i would inline that variable, it doesn't add much value for me.

Reply:
sure